### ISSUE TITLE
DistilUSE CoreML variant: forward pass fails on iOS

### ISSUE THREAD

**Issue by msluszniak** [software-mansion/react-native-executorch#1164]
### Description

The CoreML build of `distiluse-base-multilingual-cased-v2` downloads successfully from HuggingFace but its first `forward()` invocation fails at the C++ layer on iOS. Surfaces as:

```
[TextEmbeddings.cpp] Forward pass failed (in: std::shared_ptr<OwningArrayBuffer>
rnexecutorch::models::embeddings::TextEmbeddings::generate(const std::string)).
Ensure the model input is correct.
```

This is reproducible on both `main` and PR #1148 (model-registry) — not introduced by the registry refactor.

### Reproduction

1. Run the `text-embeddings` demo app on the iOS simulator.
2. Select `Multilingual DistilUSE` (with `{ backend: 'coreml' }` before this PR; the registry variant was removed in #1148 in response).
3. Provide single word input to the example! More than one word seems to be working correctly.
4. Model loads ("Model is ready"), but the inference throws.
### Notes

- The xnnpack variant of the same model works correctly on both iOS and Android, so the tokenizer/input pipeline is fine; the failure is specific to the CoreML `.pte`.
- HuggingFace serves the file at the new `distiluse_base_multilingual_cased_v2_coreml_fp32.pte` path (200 OK); the broken URLs on `main` (hyphen form) 404 entirely, masking this bug as a download failure rather than a runtime one.
- PR #1148 drops the CoreML variant from the registry (and removes the deprecated `DISTILUSE_BASE_MULTILINGUAL_CASED_V2_COREML` export) so users only get the working xnnpack path. Re-add the variant once the CoreML export is fixed.
- Related: clearer surfacing of native-layer errors in the demo apps is tracked in #1150 — once that lands the underlying `[object Object]` problem we hit while diagnosing this won't recur.

**Comment by msluszniak** [comment]
One very important note, this breaks only when there is single word provided, word + space + word works correctly.

**Comment by chmjkb** [comment]
I think that some words work, some don't. For example: 
- [space]Asd - works
- [space]As - doesn't work

**Comment by msluszniak** [comment]
Root cause found, plus a trap in the obvious workaround. Summary: the Core ML model rejects a sequence length of 1, and padding to 2 is **not** a safe fix.

### Why it fails

`coreml/export.py` declares `Dim("tokens", min=1, max=126)`, but the compiled Core ML model enforces `2..126`, because `torch.export` specializes size 1 away. Verified against the published `v0.10.0` artifact on host (macOS, ExecuTorch 1.4.1):

```
seq=1                            FAIL  Core ML: Size (1) of dimension (1) is not in allowed range (2..126)
seq=2,3,4,5,6,8,16,32,64,126     OK, finite, correct (1, 512) output
```

That is exactly the reported symptom. `_distiluse_model.py` adds CLS/SEP itself, so a single word reaches `forward()` as `input_ids` of length 1 and is rejected, while two or more words are fine.

This is not specific to distiluse. Every published Core ML text embedder fails at `seq=1` and works from `seq=2`:

| model | seq=1 | seq=2 |
| --- | --- | --- |
| `all_minilm_l6_v2` | FAIL | OK |
| `clip_vit_base_patch32_text` | FAIL | OK |
| `multi_qa_minilm_l6_cos_v1` | FAIL | OK |
| `paraphrase_multilingual_minilm_l12_v2` | FAIL | OK |
| `distiluse_base_multilingual_cased_v2` | FAIL | OK |

The Core ML build is otherwise numerically fine: cosine against the XNNPACK fp32 build is 0.99998 at seq 2, 4, 8 and 16, so this is purely a shape restriction, not a precision or correctness problem.

### Why "just pad to 2 tokens" does not work

The natural workaround is to pad short inputs and mask the padding, but that silently corrupts the embedding. Measured on the XNNPACK build, so it is not a Core ML effect:

```
1 real token, vs the same token padded to 2 and masked:   cosine 0.89 to 0.97
4 real tokens + 4 masked pads:                            cosine 0.746
8 real tokens + 8 masked pads:                            cosine 0.635
16 real tokens + 16 masked pads:                          cosine 0.426
```

The reason is in `_distiluse_model.py`: `attention_mask` is used for the mean pool, but line 32 passes `attention_mask=None` into BERT, deliberately, with the comment "RNE never pads, so it is all-ones". Padding therefore participates in self-attention and corrupts the real tokens' hidden states before pooling ever runs. Anyone adding padding today gets a wrong embedding with no error.

### Suggested fix

Wire `attention_mask` through to BERT rather than passing `None`:

```python
tokens = self.bert(input_ids=input_ids, attention_mask=attention_mask)[0]
```

For unpadded input the mask is all ones, which is what BERT already assumes, so this is a no-op for every current caller and the artifacts stay numerically equivalent. With it in place, padding becomes valid, and the `seq=1` case can be handled by padding to 2 on either side of the boundary. This fixes all five models with one change and needs no Core ML specific workaround.

Alternative, if the export should accept `seq=1` directly: it needs a shape representation `torch.export` will not specialize, which is more invasive for no extra benefit given the above.

### Meanwhile

The Core ML variant stays unwired in #1375, per the note in this issue to re-add once the export is fixed. The MLX int8 variant is unaffected and is wired.

**Comment by msluszniak** [comment]
Completed, re-exported models are now live
